Fix `stm32f412g_disco` building issue on `samples/drivers/flash_shell`

```
[134/176] Building C object modules/stm32/stm32cube/CMake...32cube.dir/stm32f4xx/drivers/src/stm32f4xx_hal_qspi.c.obj
~/zephyrproject/modules/hal/stm32/stm32cube/stm32f4xx/drivers/src/stm32f4xx_hal_qspi.c: In function 'HAL_QSPI_Command_IT':
~/zephyrproject/modules/hal/stm32/stm32cube/stm32f4xx/drivers/src/stm32f4xx_hal_qspi.c:873:12: warning: unused variable 'tickstart' [-Wunused-variable]
  873 |   uint32_t tickstart = HAL_GetTick();
      |            ^~~~~~~~~

```

--- a/west.yml
+++ b/west.yml
@@ -114,7 +114,7 @@ manifest:
       groups:
         - hal
     - name: hal_stm32
-      revision: ac47229a9809a6d2250c14ff593e1b29625620be
+      revision: 2d95b36c57c3245e2d08714592790750c95a73af
       path: modules/hal/stm32
       groups:
         - hal

